src/CameraSensor.cc

```diff
@@ -241,6 +337,33 @@ bool CameraSensor::CreateCamera()
     this->dataPtr->saveImage = true;
   }
 
+  // Update the DOM object intrinsics to have consistent
+  // projection matrix values between ogre camera and camera_info msg
+  // If these values are not defined in the SDF then we need to update
+  // these values to something reasonable. The projection matrix is
+  // the cumulative effect of intrinsic and extrinsic parameters
+  if(!cameraSdf->HasLensProjection())
+  {
+    // Note that the matrix from Ogre via camera->ProjectionMatrix() has a
+    // different format than the projection matrix used in SDFormat.
+    // This is why they are converted using projectionToCameraIntrinsic.
+    // The resulting matrix is the intrinsic matrix, but since the user has
+    // not overridden the values, this is also equal to the projection matrix.
+    auto intrinsicMatrix =
+      gz::rendering::projectionToCameraIntrinsic(
+        this->dataPtr->camera->ProjectionMatrix(),
+        this->dataPtr->camera->ImageWidth(),
+        this->dataPtr->camera->ImageHeight()
+      );
+    cameraSdf->SetLensProjectionFx(intrinsicMatrix(0, 0));
+    cameraSdf->SetLensProjectionFy(intrinsicMatrix(1, 1));
+    cameraSdf->SetLensProjectionCx(intrinsicMatrix(0, 2));
+    cameraSdf->SetLensProjectionCy(intrinsicMatrix(1, 2));
+  }
```

Comment:
also port this [else](https://github.com/gazebosim/gz-sensors/blob/7b4387fc64b12f72a39e7c7efafcbc8cc4230766/src/CameraSensor.cc#L368-L385) block?

Reply:
https://github.com/gazebosim/gz-sensors/pull/383/commits/f346f233ff8fc9488f78dc4c3fd7aadc61a3c815 and https://github.com/gazebosim/gz-sensors/pull/383/commits/582d9d169ba7b5f9064f8bebf4f4fa519433ea2f